src/Customer/Types/TrackCustomerEvent.hs

```diff
@@ -0,0 +1,26 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+module Customer.Types.TrackCustomerEvent
+  ( TrackCustomerEventBody(..)
+  , defaultTrackCustomerEvent
+  ) where
+
+import Customer.Aeson (defaultAesonOptions)
+import Data.Aeson (Object, ToJSON(toJSON), genericToJSON)
+import Data.Text (Text)
+import GHC.Generics (Generic)
+
+data TrackCustomerEventBody = MkTrackCustomerEvent
+  { tceName      :: Text
+  , tceId        :: Maybe Text
+  , tceTimestamp :: Maybe Int
+  , tceData      :: Maybe Object
```

Comment:
AFAICS, this API endpoint allows any additional payload in JSON format. How hard will it be to make it polymorphic and allow providing an object of type `a`, that has `ToJSON` instance?

Reply:
Seems like it is not possible due to necessity to provide ToJSON instance to servant's `client` function. I can make it existential, but there would be problems with making right ToJSON instance.